agents/community/aider/src/aider_agent/agent.py

```diff
@@ -5,23 +5,145 @@
 import sys
 import tempfile
 from pathlib import Path
+from collections.abc import AsyncGenerator
+from textwrap import dedent
 
-from pydantic import Field
-from beeai_sdk.schemas.text import TextInput, TextOutput
-from beeai_sdk.providers.agent import Server, Context
+from acp_sdk import Message, Metadata, Link, LinkType, MessagePart, Artifact
+from acp_sdk.server import Context, Server
+from pydantic import AnyUrl
+import mimetypes
 
+server = Server()
 
-class Output(TextOutput):
-    files: dict[str, str] = Field(default_factory=dict)
-    text: str = Field(default_factory=str)
 
+@server.agent(
+    name="aider",
+    description="An AI pair programmer that edits code in a local Git repository using natural language, executing commands and providing feedback.",
+    metadata=Metadata(
+        programming_language="Python",
+        links=[
+            Link(
+                type=LinkType.SOURCE_CODE,
+                url=AnyUrl(
+                    f"https://github.com/i-am-bee/beeai-platform/blob/{os.getenv('RELEASE_VERSION', 'main')}"
+                    "/agents/community/aider"
+                ),
+            )
+        ],
+        license="Apache 2.0",
+        framework="Custom",
+        documentation=dedent(
+            """
```

Comment:
This won't add newline at the beginning:
```suggestion
            """\
```